.github/workflows/update-schedule.yml

```diff
@@ -0,0 +1,46 @@
+name: Update schedule.yaml
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 1'  # Every monday
+jobs:
+  create-pull-request:
+    name: Create PR (if required)
+    if: github.repository == 'kubernetes/website'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          check-latest: true
+
+      - name: Install schedule-builder
+        run: go install k8s.io/release/cmd/schedule-builder@latest
+
+      - name: Update schedule.yaml
+        run: schedule-builder -uc data/releases/schedule.yaml
+
+      - name: Check workspace
+        id: create_pr
+        run: |
+          if [[ $(git diff --stat) != '' ]]; then
+            echo "create_pr=true" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v6
```

Comment:
Is this pin (to `v6`) enough to manage supply chain risks?

Reply:
I can change them to pin to the commit sha if that helps.

Reply:
please always try to use commit sha and add a comment of which version

like 
```
uses: peter-evans/create-pull-request@dscvdscvdsc324234 # v6.y.z
```